From ffef9040f0fc39576553b2f27f3294ae9fa600bd Mon Sep 17 00:00:00 2001
From: Noam.Lessner <nlessner@habana.ai>
Date: Wed, 01 Sep 2021 21:09:19 +0300
Subject: [PATCH] [SW-0] gaudi2: add pre poweron flag

Add flag to disable all code copied from LKD poweron.
for pre-poweron this will not run, only when poweron will
be ready to start integration commenting out or deleting this
commit will enable these code section.

Change-Id: I9ba97571340ed2a6581fea7a98712f84a447bf58
---

diff --git a/arch/gaudi2/common/axuser.c b/arch/gaudi2/common/axuser.c
index 48bacb2..bbd7bf4 100644
--- a/arch/gaudi2/common/axuser.c
+++ b/arch/gaudi2/common/axuser.c
@@ -520,6 +520,11 @@
 
 void preboot_axuser_override(void)
 {
+#ifdef PRE_POWER_ON
+	/* if we're pre poweron don't run this code yet */
+	return;
+#endif
+
 	gaudi2_init_mstr_if(mmPCIE_MSTR_RR_MSTR_IF_AXUSER_BASE);
 	gaudi2_init_dma_core_axuser_overrides();
 	gaudi2_init_qm_axuser_overrides();
diff --git a/arch/gaudi2/common/common.h b/arch/gaudi2/common/common.h
index dbf85a7..79eaa5f 100644
--- a/arch/gaudi2/common/common.h
+++ b/arch/gaudi2/common/common.h
@@ -9,6 +9,9 @@
 #include "credits.h"
 #include "axuser.h"
 
+/* pre power-on flag, comment out to enable FW config insted LKD power-on */
+#define PRE_POWER_ON
+
 #define GAUDI2_DECODER_FULL_MASK	0x3FF
 #define GAUDI2_EDMA_FULL_MASK		0xFF
 #define GAUDI2_TPC_FULL_MASK		0x1FFFFFF
diff --git a/arch/gaudi2/common/credits.c b/arch/gaudi2/common/credits.c
index 0c3c2d1..639deab 100644
--- a/arch/gaudi2/common/credits.c
+++ b/arch/gaudi2/common/credits.c
@@ -513,6 +513,10 @@
 
 void preboot_e2e_credits_init(void)
 {
+#ifdef PRE_POWER_ON
+	/* if we're pre poweron don't run this code yet */
+	return;
+#endif
 
 	/* enable PCIe E2E credits */
 	regmap_writel(0, mmPMMU_PIF_DISABLE_E2E_CREDITS);
